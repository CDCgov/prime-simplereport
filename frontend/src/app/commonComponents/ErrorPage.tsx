import USAGovBanner from "../commonComponents/USAGovBanner";
import siteLogo from "../../img/simplereport-logo-color.svg";

const ErrorPage = () => {
  return (
    <>
      <USAGovBanner />
      <header className="border-bottom border-base-lighter padding-y-1">
        <div className="grid-container">
          <a href={import.meta.env.VITE_PUBLIC_URL || "/"}>
            <img
              className="maxh-4"
              src={siteLogo}
              alt={import.meta.env.VITE_TITLE}
            />
          </a>
        </div>
      </header>
      <main className="usa-section">
        <div className="grid-container usa-prose">
          <h1 className="">Something went wrong :(</h1>
          <p className="usa-intro">Please try refreshing your browser.</p>
          <p>
            If the problem continues, contact{" "}
            <a href="mailto:support@simplereport.gov">
              support@simplereport.gov
            </a>{" "}
            for support.
          </p>
        </div>
      </main>
    </>
  );
};

export default ErrorPage;
