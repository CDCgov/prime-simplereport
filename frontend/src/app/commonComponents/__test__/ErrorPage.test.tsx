import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import ErrorPage from "../ErrorPage";

describe("ErrorPage", () => {
  it("displays the correct error message", () => {
    render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    );
    expect(screen.getByText("Something went wrong :(")).toBeInTheDocument();
    expect(
      screen.getByText("Please try refreshing your browser.")
    ).toBeInTheDocument();
    expect(screen.getByText("support@simplereport.gov")).toHaveAttribute(
      "href",
      "mailto:support@simplereport.gov"
    );
  });
});
