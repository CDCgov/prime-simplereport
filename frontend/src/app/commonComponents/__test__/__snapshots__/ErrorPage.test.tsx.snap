// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage displays the correct error message 1`] = `
<div>
  <div
    class="usa-banner"
  >
    <div
      class="usa-accordion"
    >
      <header
        class="usa-banner__header"
      >
        <div
          class="usa-banner__inner"
        >
          <div
            class="grid-col-auto"
          >
            <img
              alt="U.S. flag"
              class="usa-banner__header-flag"
              src="us_flag_small.png"
            />
          </div>
          <div
            class="grid-col-fill tablet:grid-col-auto"
          >
            <p
              class="usa-banner__header-text"
            >
              An official website of the United States government
            </p>
            <p
              aria-hidden="true"
              class="usa-banner__header-action"
            >
              Here’s how you know
            </p>
          </div>
          <button
            aria-controls="gov-banner"
            aria-expanded="false"
            class="usa-accordion__button usa-banner__button"
          >
            <span
              class="usa-banner__button-text"
            >
              Here’s how you know
            </span>
          </button>
        </div>
      </header>
    </div>
  </div>
  <header
    class="border-bottom border-base-lighter padding-y-1"
  >
    <div
      class="grid-container"
    >
      <img
        alt="{process.env.REACT_APP_TITLE}"
        class="maxh-4"
        src="simplereport-logo-color.svg"
      />
    </div>
  </header>
  <main
    class="usa-section"
  >
    <div
      class="grid-container usa-prose"
    >
      <h1
        class=""
      >
        Something went wrong :(
      </h1>
      <p
        class="usa-intro"
      >
        Please try refreshing your browser.
      </p>
      <p>
        If the problem continues, contact
         
        <a
          href="mailto:support@simplereport.gov"
        >
          support@simplereport.gov
        </a>
         
        for support.
      </p>
    </div>
  </main>
</div>
`;
